server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataQuery.java

```diff
@@ -394,24 +423,44 @@ private UnmodifiableIterator<DataSegment> retrieveSegmentsInIntervalsBatch(
       final Collection<Interval> intervals,
       final IntervalMode matchMode,
       final boolean used,
-      @Nullable final Integer limit
+      @Nullable final Integer limit,
+      @Nullable final String lastSegmentId,
+      @Nullable final SortOrder sortOrder
   )
   {
     // Check if the intervals all support comparing as strings. If so, bake them into the SQL.
     final boolean compareAsString = intervals.stream().allMatch(Intervals::canCompareEndpointsAsStrings);
 
     final StringBuilder sb = new StringBuilder();
-    sb.append("SELECT payload FROM %s WHERE used = :used AND dataSource = :dataSource");
+    sb.append("SELECT payload FROM %s WHERE used = :used AND dataSource = :dataSource %s");
 
     if (compareAsString) {
       appendConditionForIntervalsAndMatchMode(sb, intervals, matchMode, connector);
     }
 
```

Comment:
I think appending the id here would make it less error-prone and keeps the sql query construction flow:
```suggestion
    if (lastSegmentId != null) {
        sb.append(
            StringUtils.format("AND id %s :id",
                               (sortOrder == null || sortOrder == SortOrder.ASC)
                               ? ">"
                               : "<")
        );
```

Reply:
fixed